### ISSUE TITLE
nested stack nuance - potential issue

### ISSUE THREAD

**Issue by louisholley** [expo/router#79]
hey, i'm not sure if this is an issue or whether it's something wrong with my setup but this is my file setup

```
app/
  (home).tsx
  (home)/
    index.tsx
    inbox/
      index.tsx
      [id].tsx
```

(home).tsx is just a stack like this
```
import { Stack } from "expo-router";

export default Stack;
```

(home)/index.tsx renders a custom header with `<Stack.Screen ...` which is working fine

however trying to render the same custom header with `Stack.Screen` within inbox/index or inbox/[id] doesn't work - it renders the default react navigation header. even if i add a layout root (home)/inbox.tsx with another stack like `<Stack screenOptions={{headerShown:false}} />`, the default header is rendered for the inbox screens

any ideas? no clue how to proceed!

**Comment by EvanBacon** [comment]
I'm going to need a minimum repro, the written description is a bit hard to follow.

I'd imagine the issue is related to the virtual navigator at `app/(home)/inbox/`, to support React Navigation a virtual file `app/(home)/inbox.js` is created (functionality [documented here](https://expo.github.io/router/docs/features/layout-routes)). So the nested screen will need to set the options for its [grandparent navigator](https://github.com/expo/router/pull/48).

**Comment by louisholley** [comment]
hey @EvanBacon setting options for the grandparent navigator sorted it - thanks so much! :)
